From 33f4291f00d0d89e57fbcb133176880e23698c55 Mon Sep 17 00:00:00 2001
From: "sheng.liu" <sheng.liu@amlogic.com>
Date: Fri, 31 Dec 2021 18:18:05 +0800
Subject: [PATCH] gstreamer1.0-libav: Increase the MAX_STREAMS  [1/1]

PD#SWPL-65566

Problem:
The test file have 25 streams, but ffmpeg found the number is 37
We can not modify the ffmpeg, so increate the MAX_STREAMS

Solution:
Increase the number of MAX_STREAMS to 40

Verify:
AP222
---
 ext/libav/gstavdemux.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/ext/libav/gstavdemux.c b/ext/libav/gstavdemux.c
index 0edc5e2..f3cc93b 100644
--- a/ext/libav/gstavdemux.c
+++ b/ext/libav/gstavdemux.c
@@ -36,7 +36,7 @@
 #include "gstavutils.h"
 #include "gstavprotocol.h"
 
-#define MAX_STREAMS 20
+#define MAX_STREAMS 40
 
 typedef struct _GstFFMpegDemux GstFFMpegDemux;
 typedef struct _GstFFStream GstFFStream;
@@ -1254,6 +1254,11 @@ gst_ffmpegdemux_open (GstFFMpegDemux * demux)
 
   n_streams = demux->context->nb_streams;
   GST_DEBUG_OBJECT (demux, "we have %d streams", n_streams);
+  if (n_streams > MAX_STREAMS)
+  {
+    GST_ERROR_OBJECT (demux, "The n_streams %d exceeds the MAX_STREAMS(40) ", n_streams);
+    goto beach;
+  }
 
   /* open_input_file() automatically reads the header. We can now map each
    * created AVStream to a GstPad to make GStreamer handle it. */
-- 
2.32.0

